### PR TITLE
Add parameters for Microsoft Account Authentication

### PR DESCRIPTION
**Pull request by Kahbazi**
#11026

### CODE REVIEW

**Comment by Tratcher** [comment]
Merege conflict? Please rebase.

File: src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs [hunk 1]
```diff
@@ -38,5 +41,60 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIden
                 return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
             }
         }
+
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
```
Comment (Tratcher) [inline src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs]: BuildChallengeUrl now has additional functionality you should copy: https://github.com/aspnet/AspNetCore/blob/c89055b9b0664ca926a2ad7305969777ef32d319/src/Security/Authentication/OAuth/src/OAuthHandler.cs#L261-L276

Comment (Kahbazi) [reply src/Security/Authentication/MicrosoftAccount/src/MicrosoftAccountHandler.cs]: Done

File: src/Security/Authentication/test/MicrosoftAccountTests.cs [hunk 2]
```diff
@@ -244,6 +244,36 @@ public async Task AuthenticatedEventCanGetRefreshToken()
             Assert.Equal("Test Refresh Token", transaction.FindClaimValue("RefreshToken"));
         }
 
+        [Fact]
+        public async Task ChallengeWillUseAuthenticationPropertiesParametersAsQueryArguments()
+        {
+            var stateFormat = new PropertiesDataFormat(new EphemeralDataProtectionProvider(NullLoggerFactory.Instance).CreateProtector("GoogleTest"));
```
Comment (campersau) [inline src/Security/Authentication/test/MicrosoftAccountTests.cs]:
```suggestion
            var stateFormat = new PropertiesDataFormat(new EphemeralDataProtectionProvider(NullLoggerFactory.Instance).CreateProtector("MicrosoftTest"));
```

File: src/Security/Authentication/MicrosoftAccount/src/MicrosoftChallengeProperties.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Authentication.OAuth;
+
+namespace Microsoft.AspNetCore.Authentication.MicrosoftAccount
+{
+    public class MicrosoftChallengeProperties : OAuthChallengeProperties
```
Comment (Tratcher) [inline src/Security/Authentication/MicrosoftAccount/src/MicrosoftChallengeProperties.cs]: Please link to the doc that defines each of these keys.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**